### PR TITLE
Add get_rows_by_tags

### PR DESCRIPTION
**Pull request by tracivar**
- `self.tags` is dict with keys as tag names and values as an list of indicies.
- `get_rows_by_tags` receives a list of tag names and returns rows of metadata which has tags in the tag names.
- tags are cached by default

https://zegami.atlassian.net/browse/ZGM-7419

### CODE REVIEW

**Review by tarqu1n — APPROVED** [review]

File: zegami_sdk/collection.py [hunk 1]
```diff
@@ -175,6 +177,19 @@ def rows(self):
...
+        pass
+
+    @tags.getter
+    def tags(self):
+        if self.allow_caching and self._cached_tags is not None:
+            return self._cached_tags
+        tags = self.get_tags()
+        if self.allow_caching:
+            self._cached_tags = tags
+        return tags
```
Comment (Vaspra) [inline zegami_sdk/collection.py:192]: Given that tags are intended to be a sort've dynamic column, I wonder if caching may be problematic compared to rows. If a user modifies the tags and doesn't clear the cache, they will get incorrect data.

File: zegami_sdk/collection.py [hunk 2]
```diff
@@ -259,6 +289,23 @@ def download_single(index, url):
         for i in range(len(images)):
             ordered.append(images[i])
         return ordered
+    
+    def get_tags(self):
```
Comment (Vaspra) [inline zegami_sdk/collection.py:293]: Could this be hidden as `_get_tags`? Thinking from an outsider, they wouldn't know what to do with tag indices.

Comment (ahaith) [reply zegami_sdk/collection.py:293]: Or better still `_get_tag_indices`?

File: zegami_sdk/collection.py (hunk 1 above)
Comment (tracivar) [reply zegami_sdk/collection.py:192]: Is it better to not cache the tags? I was trying to mimic the same behaviour in js-sdk where tags are cached.

**Pull request merged**